### DISCUSSION TITLE
Forcefully disable any Javascript

### DISCUSSION THREAD

**Discussion by lcl45** [mozilla/pdf.js#18428] [Q&A]
Hi all!

I'm struggling with the pdfjsLib. What I'm trying to do is to forcefully disable any Javascript present inside a pdf: all the pdf I'm going to show in the application I'm working in must be some kind of read-only file, preventing any possible code execution, even if benign.
I know that this might sound strange because "this could break the pdf" or something like that, but this was requested.

In the application, we are using the default viewer found in `web/viewer.html`, calling it using an url similar to `web/viewer.html?file=my.pdf`. I've found that there is the `enableScripting` variable which might help with my issue, however I've found it here https://github.com/mozilla/pdf.js/wiki/Viewer-options that it is not a valid parameter.

In the main script of the application, I've added
```javascript
document.addEventListener('DOMContentLoaded', function () {
    if (window.PDFViewerApplicationOptions) {
          PDFViewerApplicationOptions.set("disablePreferences", true);
          PDFViewerApplicationOptions.set("enableScripting", false);
          PDFViewerApplicationOptions.set("isEvalSupported", false);
          PDFViewerApplication.run();
    }
});
```
This doesn't seem to trigger the warning that should pop out (like discussed here https://github.com/mozilla/pdf.js/issues/16158). 
How can I achive my goal?

Many thanks!!

**Comment by bubakcz** [comment]
Hello. 
I have a similar issue. 
In our code base and repo, we have currently available version 4.4.168 of this library. With unchanged viewer js and slightly modified viewer html, even with same attributes as OP, test pdf is still running included javascript (`<script>app.alert(1);</script>`). Are there any changes required to default viewer script, or could be just some issue with that version and we should use a newer one?

**Reply by bubakcz** [reply]
Tried it also with recent version of library, and javascript is still being executed. This happens both with PDF displayed via "file" parameter, and PDFViewerApplication.open method for Uint8Array approach.
Any ideas?

**Comment by calixteman** [comment]
You should set the options you want enable/disable before the setting the different components, please read:
https://github.com/mozilla/pdf.js/blob/91384738f8df68cee668b285d6d6a778992d3b4d/web/viewer.js#L299-L323
